src/process_impl/mod.rs

```diff
@@ -0,0 +1,203 @@
+#![allow(
+    missing_docs,
+    missing_debug_implementations,
+    rustdoc::broken_intra_doc_links,
+    unreachable_pub
+)]
+
+use std::ffi::OsStr;
+use std::io;
+use tokio::io::AsyncReadExt;
+use tokio::process;
+
+use super::Error;
+
+use super::{ForwardType, Socket};
+
+pub(crate) mod builder;
+
+mod command;
+pub(crate) use command::Command;
+
+mod child;
+pub(crate) use child::RemoteChild;
+
+#[derive(Debug)]
+pub(crate) struct Session {
+    ctl: tempfile::TempDir,
+    addr: String,
+    terminated: bool,
+    master: std::sync::Mutex<Option<(tokio::process::ChildStdout, tokio::process::ChildStderr)>>,
+}
+
+impl Session {
```

Comment:
Has anything changed in the body of this `impl` block from what used to be there before this PR?

Reply:
There are two modifications to it:
 - implementation of `request_port_forwarding`;
 - removal of `command` and `shell` since these two functions are shared between `process_impl` and `mux_client_impl`.